rust/src/connection/network/transmitter/transaction.rs

```diff
@@ -241,60 +242,46 @@ impl TransactionTransmitter {
             on_close: Default::default(),
             callback_handler_sink,
         };
-        tokio::spawn(Self::dispatch_loop(
-            queue_source,
-            request_sink,
-            collector.clone(),
-            on_close_callback_source,
-            shutdown_signal,
-        ));
+        let cloned_collector = collector.clone();
+        tokio::task::spawn_blocking(move || {
+            Self::dispatch_loop(queue_source, request_sink, cloned_collector, on_close_callback_source, shutdown_signal)
+        });
         tokio::spawn(Self::listen_loop(response_source, collector, shutdown_sink));
     }
 
-    async fn dispatch_loop(
+    fn dispatch_loop(
         mut request_source: UnboundedReceiver<(TransactionRequest, Option<ResponseSink<TransactionResponse>>)>,
         request_sink: UnboundedSender<transaction::Client>,
         mut collector: ResponseCollector,
         mut on_close_callback_source: UnboundedReceiver<Box<dyn FnOnce(Option<Error>) + Send + Sync>>,
         mut shutdown_signal: UnboundedReceiver<()>,
     ) {
         const MAX_GRPC_MESSAGE_LEN: usize = 1_000_000;
-        const DISPATCH_INTERVAL: Duration = Duration::from_millis(3);
+        const DISPATCH_INTERVAL: Duration = Duration::from_micros(10);
```

Comment:
Have you considered nixing this altogether and just sending the requests eagerly? That would remove the sleeps without us having to spawn another thread.